components/core/src/streaming_archive/writer/Archive.hpp

```diff
@@ -68,57 +69,50 @@ namespace streaming_archive { namespace writer {
          * @param user_config Settings configurable by the user
          * @throw FileWriter::OperationFailed if any dictionary writer could not be opened
          * @throw streaming_archive::writer::Archive::OperationFailed if archive already exists, if it could not be stat-ed, if the directory structure could
-                  not be created or problems with medatadata.
+                  not be created, if the file is not reset or problems with medatadata.
          */
         void open (const UserConfig& user_config);
         /**
          * Writes a final snapshot of the archive, closes all open files, and closes the dictionaries
          * @throw FileWriter::OperationFailed if any writer could not be closed
          * @throw streaming_archive::writer::Archive::OperationFailed if any empty directories could not be removed
+         * @throw streaming_archive::writer::Archive::OperationFailed if the file is not reset
          * @throw Same as streaming_archive::writer::SegmentManager::close
          * @throw Same as streaming_archive::writer::Archive::write_dir_snapshot
          */
         void close ();
 
         /**
-         * Creates a file with the given path
+         * Creates and opens a file with the given path
          * @param path
          * @param group_id
          * @param orig_file_id
          * @param split_ix
          * @return Pointer to the new file
          */
-        File* create_file (const std::string& path, group_id_t group_id, const boost::uuids::uuid& orig_file_id, size_t split_ix);
+        void create_and_open_file (const std::string& path, group_id_t group_id, const boost::uuids::uuid& orig_file_id, size_t split_ix);
 
-        /**
-         * Wrapper for streaming_archive::writer::File::open
-         * @param file File to open
-         * @throw Same as streaming_archive::writer::File::open
-         */
-        void open_file (File& file);
         /**
          * Wrapper for streaming_archive::writer::File::close
-         * @param file File to close
          * @throw Same as streaming_archive::writer::File::close
          */
-        void close_file (File& file);
-        bool is_file_open (File& file);
+        void close_file ();
```

Comment:
To clean-up the code, instead of all these wrappers, how about we add a method `File& get_file () const { return *m_file; }` and then call the methods on the file itself externally?